src/MongoDB.Driver.Core/Core/Operations/RetryableUpdateCommandOperation.cs

```diff
@@ -180,14 +180,22 @@ protected override void SerializeValue(BsonSerializationContext context, BsonSer
             }
 
             // private methods
+            private IBsonSerializer GetUpdateSerializer(BsonValue value)
+            {
+                return value.IsBsonArray
+                    ? (IBsonSerializer)BsonArraySerializer.Instance
+                    : BsonDocumentSerializer.Instance;
+            }
+
             private void SerializeUpdate(BsonSerializationContext context, BsonSerializationArgs args, UpdateRequest request)
             {
                 var writer = context.Writer;
                 writer.PushElementNameValidator(ElementNameValidatorFactory.ForUpdateType(request.UpdateType));
                 try
                 {
                     var position = writer.Position;
-                    BsonDocumentSerializer.Instance.Serialize(context, request.Update);
+                    var serializer = GetUpdateSerializer(request.Update);
+                    serializer.Serialize(context, request.Update);
```

Comment:
I think the previous two lines can be simplified to:

     BsonValueSerializer.Instance.Serialize(context, request.Update);